resources/sdk/clisdkclient/extensions/cmd/profiles/createprofile.go

```diff
@@ -77,35 +104,100 @@ func requestUserInput() config.Configuration {
 		environment = "mypurecloud.com"
 	}
 
-	fmt.Printf("Please provide either an Access Token, Client Credentials (Client ID and Client Secret) or both\n")
-	fmt.Printf("Note: If you provide an Access Token, this will be used over Client Credentials\n")
-	for {
-		fmt.Printf("Access Token: ")
-		fmt.Scanln(&accessToken)
-		fmt.Printf("Client ID: ")
-		fmt.Scanln(&clientID)
-		fmt.Printf("Client Secret: ")
-		fmt.Scanln(&clientSecret)
-		// users should only be allowed to continue if they provide an access token
-		if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) == 0 && len(strings.TrimSpace(clientSecret)) == 0 {
-			clientID = ""
-			clientSecret = ""
-			break
-			// or if they provide client credentials
-		} else if len(strings.TrimSpace(accessToken)) == 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
-			accessToken = ""
-			break
-			// or both
-		} else if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
+	fmt.Print("Note: If you provide an access token, this will take precedence over any authorization grant type.\n")
+	fmt.Print("Access Token (Optional): ")
+	fmt.Scanln(&accessToken)
+
+	for true {
+		fmt.Print("Select your authorization grant type.\n")
+		fmt.Print("\t0. None\n\t1. Client Credentials\n\t2. Implicit Grant\nGrant Type: ")
+		fmt.Scanln(&grantType)
+
+		if accessToken == "" && grantType == None {
+			fmt.Print("If you have not provided an access token, you must select a grant type.\n")
+			continue
+		}
+		if isValidGrantType(grantType) {
 			break
-			// otherwise
+		}
+	}
+
+	clientID, clientSecret = requestClientCreds(accessToken, grantType)
+
+	if grantType == ImplicitGrant {
+		for true {
+			fmt.Print("Would you like to use a secure HTTP connection? [Y/N]: ")
+			fmt.Scanln(&authChoice)
+			if strings.ToUpper(authChoice) == "Y" {
+				secureLoginEnabled = true
+				break
+			} else if strings.ToUpper(authChoice) == "N" {
+				secureLoginEnabled = false
+				break
+			}
+		}
+		redirectURI = requestRedirectURI(secureLoginEnabled)
+		fmt.Printf("Redirect URI: %s\n", redirectURI)
```

Comment:
Print this out after they select whether they want a secure connection or not. Then it will be correct (http or https)

Reply:
The function `requestRedirectURI` will return the string with "http" or "https" depending on the value in the input param. Do you mean I should ask for the port, then if they want a secure connection, then print it out?

Reply:
Yep, so it would look like this:
```
Profile Name [DEFAULT]: 
Environment [mypurecloud.com]: a
Note: If you provide an access token, this will take precedence over any authorization grant type.
Access Token (Optional): 
Select your authorization grant type.
        0. None
        1. Client Credentials
        2. Implicit Grant
Grant Type: 2
Client ID: a
Client Secret (Optional): 
Redirect URI port [8080]: 
Would you like to open a secure HTTP connection? [Y/N]: Y
Redirect URI: https://localhost:8080
Profile name DEFAULT already exists in the config file. Overwrite (Y/N): 
```